app/furniture/marketplace/square_idempotency_key.rb

```diff
@@ -0,0 +1,7 @@
+class Marketplace
+  class SquareIdempotencyKey
+    def self.generate(order_id)
+      "#{order_id}_#{8.times.map { rand(10) }.join}"
```

Comment:
You can also use `SecureRandom.hex(8)` to generate 8 random hex characters

Reply:
Oh, me likey.